Source/Core/Core/Config/DefaultLocale.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/Config/DefaultLocale.h"
+
+#include <algorithm>
+#include <array>
+#include <locale>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif
+
+#include "Common/Assert.h"
+#include "Common/CommonTypes.h"
+#include "Common/StringUtil.h"
+#include "Core/Host.h"
+#include "DiscIO/Enums.h"
+
+namespace Config
+{
+static std::vector<std::string> GetPreferredLocales()
+{
+  static const std::vector<std::string> locales = Host_GetPreferredLocales();
+  return locales;
+}
+
+static std::optional<DiscIO::Language> TryParseLanguage(const std::string& locale)
+{
+  const std::vector<std::string> split_locale = SplitString(locale, '-');
+  if (split_locale.empty())
+    return std::nullopt;
+
+  // Special handling of Chinese due to its two writing systems
+  if (split_locale[0] == "zh")
+  {
+    const auto locale_contains = [&split_locale](const std::string& str) {
+      return std::find(split_locale.cbegin(), split_locale.cend(), str) != split_locale.cend();
+    };
+
+    if (locale_contains("Hans"))
+      return DiscIO::Language::SimplifiedChinese;
+    else if (locale_contains("Hant"))
+      return DiscIO::Language::TraditionalChinese;
+
+    // Mainline China and Singapore use simplified characters
+    if (locale_contains("CN") || locale_contains("SG"))
+      return DiscIO::Language::SimplifiedChinese;
+    else
+      return DiscIO::Language::TraditionalChinese;
+  }
+
+  constexpr std::array<std::string_view, 10> LANGUAGES = {
+      "ja", "en", "de", "fr", "es", "it", "nl", "zh", "zh", "ko",
```

Comment:
Might be worth mentioning that this must match the Language enum. Though it might be obvious...